http://sourceware.org/ml/binutils/2006-07/msg00029.html
http://sourceware.org/ml/binutils/2006-05/msg00319.html

From: Alan Modra <amodra at bigpond dot net dot au>
To: Etienne Lorrain <etienne_lorrain at yahoo dot fr>
Cc: binutils at sourceware dot org
Date: Fri, 19 May 2006 15:39:03 +0930
Subject: Re: `.sym' referenced in section `reloc_sym' of file.o: defined in discarded section `.text.sym' of file.o

On Wed, May 17, 2006 at 01:03:43PM +0200, Etienne Lorrain wrote:
> > > ld: error: no memory region specified for loadable section `.rel.dyn'
>
>   Looks like that is now a warning - I do not understand why.

ld/
	* ldlang.c (lang_size_sections_1): Don't check mem regions for
	os->ignored sections.
ld/testsuite/
	* ld-scripts/empty-orphan.t: Discard .reginfo.
	* ld-scripts/empty-orphan.d: Update.

===================================================================
RCS file: /cvs/src/src/ld/ldlang.c,v
retrieving revision 1.216.2.1
retrieving revision 1.216.2.2
diff -u -r1.216.2.1 -r1.216.2.2
--- src/ld/ldlang.c	2006/06/09 01:14:53	1.216.2.1
+++ src/ld/ldlang.c	2006/07/04 13:29:24	1.216.2.2
@@ -4207,7 +4207,8 @@
 		    /* If a loadable section is using the default memory
 		       region, and some non default memory regions were
 		       defined, issue an error message.  */
-		    if (!IGNORE_SECTION (os->bfd_section)
+		    if (!os->ignored
+			&& !IGNORE_SECTION (os->bfd_section)
 			&& ! link_info.relocatable
 			&& check_regions
 			&& strcmp (os->region->name,
===================================================================
RCS file: /cvs/src/src/ld/testsuite/ld-scripts/empty-orphan.d,v
retrieving revision 1.1
retrieving revision 1.1.6.1
diff -u -r1.1 -r1.1.6.1
--- src/ld/testsuite/ld-scripts/empty-orphan.d	2005/03/17 16:20:39	1.1
+++ src/ld/testsuite/ld-scripts/empty-orphan.d	2006/07/04 13:29:24	1.1.6.1
@@ -1,3 +1,6 @@
 #source: empty-orphan.s
 #ld: -T empty-orphan.t
-#error: no memory region specified for loadable section
+#readelf: -l --wide
+#...
+ +LOAD +[x0-9a-f]+ [x0]+70000000 [x0]+70000000 [x0]+(2|4|8|10|20|40|80) .*
+#pass
===================================================================
RCS file: /cvs/src/src/ld/testsuite/ld-scripts/empty-orphan.t,v
retrieving revision 1.1
retrieving revision 1.1.6.1
diff -u -r1.1 -r1.1.6.1
--- src/ld/testsuite/ld-scripts/empty-orphan.t	2005/03/17 16:20:39	1.1
+++ src/ld/testsuite/ld-scripts/empty-orphan.t	2006/07/04 13:29:24	1.1.6.1
@@ -17,5 +17,6 @@
    .text : { *(.text) } > text_mem : text_phdr
    .data : { *(.data) } > data_mem : data_phdr
    .bss : { *(.bss) } > data_mem : data_phdr
+   /DISCARD/ : { *(.reginfo) }
    /* .orphan_data is an orphan */
 }
